### PR TITLE
chore: Adds a  new step in release process that marks Jira version as released

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds a  new step in release process that marks Jira version as released

Link to any related issue(s): CLOUDP-241948

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

Example of successful log: 

```
Version released, please check all tickets are marked as done: https://jira.mongodb.org/projects/CLOUDP/versions/39020
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -68,3 +68,16 @@ jobs:
         env:
           GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  jira-release-version:
```
Comment (AgustinBettati) [inline .github/workflows/release.yml:107]: nit: I see this process is skipped during the execution if it is a pre-release, just wondering if worth leaving a comment or explicit if statement in this job definition so this behaviour is clear without having to go into the implementation

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by oarbusi — APPROVED** [review]
LGTM, agree with Agustin on making more explicit that the step is skipped if its pre-release

File: .github/workflows/release.yml (hunk 1 above)
Comment (lantoli) [reply .github/workflows/release.yml:107]: moved pre-release check to gha

**Review by oarbusi — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

File: tools/jira-release-version/main.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+const (
+	envJira       = "JIRA_API_TOKEN"
+	envVersion    = "VERSION_NUMBER"
+	projectID     = "CLOUDP"
+	jiraURL       = "https://jira.mongodb.org"
+	versionPrefix = "terraform-provider-"
+)
+
+func main() {
```
Comment (EspenAlbert) [inline tools/jira-release-version/main.go:23]: Beautiful abstraction level 👏

**Pull request merged**